polar-core/src/partial/constraints.rs

```diff
@@ -447,4 +438,199 @@ mod test {
+        let mut query =
+            polar.new_query_from_term(term!(call!("f", [partial!("a"), partial!("b")])), false);
+        let next = next_binding(&mut query)?;
+        assert_eq!(next[&sym!("a")], term!(1));
+        assert_eq!(next[&sym!("b")], term!(2));
+        Ok(())
+    }
+
+    #[test]
+    fn test_partial_in_arithmetic_op() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(r#"f(x) if x = x + 0;"#)?;
+        let mut query = polar.new_query_from_term(term!(call!("f", [partial!("a")])), false);
+        let error = query.next_event().unwrap_err();
+        assert!(matches!(error, PolarError {
+            kind: ErrorKind::Runtime(RuntimeError::Unsupported { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_method_call_on_partial() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(r#"g(x) if x.foo();"#)?;
+        let mut query = polar.new_query_from_term(term!(call!("g", [partial!("a")])), false);
+        let error = query.next_event().unwrap_err();
+        assert!(matches!(error, PolarError {
+            kind: ErrorKind::Runtime(RuntimeError::Unsupported { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_unifying_partials() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(r#"h(x, y) if x = y;"#)?;
+        let mut query =
+            polar.new_query_from_term(term!(call!("h", [partial!("a"), partial!("b")])), false);
+        let error = query.next_event().unwrap_err();
+        assert!(matches!(error, PolarError {
+            kind: ErrorKind::Runtime(RuntimeError::Unsupported { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_comparing_partials() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(r#"f(x, y) if x > y;"#)?;
+        let mut query =
+            polar.new_query_from_term(term!(call!("f", [partial!("a"), partial!("b")])), false);
+        let error = query.next_event().unwrap_err();
+        assert!(matches!(error, PolarError {
+            kind: ErrorKind::Runtime(RuntimeError::Unsupported { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_dot_lookup_with_partial_as_field() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(r#"f(x) if {}.(x);"#)?;
+        let mut query = polar.new_query_from_term(term!(call!("f", [partial!("a")])), false);
+        let error = query.next_event().unwrap_err();
+        assert!(matches!(error, PolarError {
+            kind: ErrorKind::Runtime(RuntimeError::TypeError { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_trivial_partials() -> TestResult {
+        let polar = Polar::new();
+        polar.load_str(
+            r#"f(x);
+               g(x) if false;"#,
+        )?;
+        let mut query = polar.new_query_from_term(term!(call!("f", [partial!("a")])), false);
+        let next = next_binding(&mut query)?;
+        assert_eq!(next[&sym!("a")], term!(true));
```

Comment:
I don't understand why this should be simplified to true. Related to other comment.